crap-code.kt

```diff
@@ -0,0 +1,26 @@
+package badcode.example
+class PerformanceFile {
+    val largeList = List(1000000) { it }
+    // Inefficient way to calculate the sum of elements
+    var sum = 0
+    fun calculateSum() {
+        for (i in 0 until largeList.size) {
+            sum += largeList[i]
+        }
+        // typo in println() function
+        prentln("Sum: $sum")
+        // Unnecessary object creation inside a loop
+        for (i in 0 until 100000) {
+            val tempObject = Any() // Creates a new object on each iteration
+        }
+        // Inefficient string concatenation
+        var longString = ""
+        for (i in 0 until 10000) {
+            longString += "a" // Creates a new string object on each concatenation
+        }
```

Comment:
Use `StringBuilder` for efficient string concatenation.
```suggestion
val stringBuilder = StringBuilder()
for (i in 0 until 10000) {
    stringBuilder.append("a")
}
val longString = stringBuilder.toString()
```